meson.build

```diff
@@ -0,0 +1,177 @@
+project('waltham', 'c',
+        version: '0.0.0',
+        license: ['MIT'],
+        meson_version: '>=0.41.0')
+version = meson.project_version()
+desc = 'Network IPC library designed to resemble Wayland'
+
+pkgconfig = import('pkgconfig')
+python2 = find_program('python2')
+gen_py = find_program('tools/gen.py')
+
+libwaltham_src = [
+    'src/waltham/demarshaller.h',
+    'src/waltham/marshaller.c',
+    'src/waltham/marshaller.h',
+    'src/waltham/marshaller_log.h',
+    'src/waltham/message.c',
+    'src/waltham/message.h',
+    'src/waltham/waltham-connection.c',
+    'src/waltham/waltham-connection.h',
+    'src/waltham/waltham-object.c',
+    'src/waltham/waltham-object.h',
+    'src/waltham/waltham-private.h',
+    'src/waltham/waltham-util.c',
+    'src/waltham/waltham-util.h',
+]
+
+libwaltham_headers = [
+    'src/waltham/waltham-connection.h',
+    'src/waltham/waltham-object.h',
+    'src/waltham/waltham-util.h',
+]
+
+protocols_src = [
+    'data/private.xml',
+    'data/public.xml',
+    'data/command.xml',
+]
+
+client_h = custom_target('waltham-client.h',
+                         input: ['src/waltham/header-preamble.txt', protocols_src],
+                         output: 'waltham-client.h',
+                         depend_files: 'tools/gen.py',
+                         command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'client', '-t', 'header'],
+                         install: true,
+                         install_dir: 'include/waltham')
+
+client_serialice = custom_target('client-serialice.c',
+                                 input: ['src/waltham/serial-preamble.txt', protocols_src],
+                                 output: 'client-serialice.c',
+                                 depend_files: 'tools/gen.py',
+                                 depends: client_h,
```

Comment:
I don't think `depends` here is right, because `gen.py` does not read the client header when generating this. Right?

Reply:
Indeed, oops.